test/e2e/framework/service/jig.go

```diff
@@ -764,16 +764,27 @@ func testReachabilityOverClusterIP(clusterIP string, sp v1.ServicePort, execPod
 	return nil
 }
 
-func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod) error {
+func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod, clusterIP string) error {
 	internalAddrs := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
 	externalAddrs := e2enode.CollectAddresses(nodes, v1.NodeExternalIP)
+	reIPV4 := regexp.MustCompile(e2enetwork.RegexIPv4)
+	isClusterIPV4 := reIPV4.MatchString(clusterIP)
```

Comment:
we have some helpers in utils.net for detecting the ipFamily or we can just parse it with net.ParseIP, I think that is better than the regex for consistency

Reply:
I was thinking the same thing, then saw regex was used in a couple of other funcs, so used it! But, updated it with `net.ParseIP` now.